apps/pace/app.ts

```diff
@@ -0,0 +1,217 @@
+{
+const Layout = require("Layout");
+
+const enum RunState {
+  RUNNING,
+  PAUSED
+}
+
+let state = RunState.PAUSED;
+let drawTimeout: TimeoutId | undefined;
+let lastUnlazy = 0;
+
+let lastResumeTime = Date.now();
+let splitTime = 0;
+let totalTime = 0;
+
+const splits: number[] = [];
+let splitDist = 0;
+let splitOffset = 0, splitOffsetPx = 0;
+
+let lastGPS = 0;
+const GPS_TIMEOUT_MS = 30000;
+
+const layout = new Layout({
+  type: "v",
+  c: [
+    {
+      type: "txt",
+      font: "6x8:2",
+      label: "Pace",
+      id: "paceLabel",
+      pad: 4
+    },
+    {
+      type: "txt",
+      font: "Vector:40",
+      label: "",
+      id: "pace",
+      halign: 0
+    },
+    {
+      type: "txt",
+      font: "6x8:2",
+      label: "Time",
+      id: "timeLabel",
+      pad: 4
+    },
+    {
+      type: "txt",
+      font: "Vector:40",
+      label: "",
+      id: "time",
+      halign: 0
+    },
+  ]
+}, {
+  lazy: true
+});
+
+const formatTime = (ms: number) => {
+  let totalSeconds = Math.floor(ms / 1000);
+  let minutes = Math.floor(totalSeconds / 60);
+  let seconds = totalSeconds % 60;
+  return `${minutes}:${seconds < 10 ? '0' : ''}${seconds}`;
+};
+
+const calculatePace = (time: number, dist: number) => {
+  if (dist === 0) return 0;
+  return time / dist / 1000 / 60;
+};
+
+const draw = () => {
+  if (state === RunState.PAUSED) {
+    // no draw-timeout here, only on user interaction
+    drawSplits();
+    return;
+  }
+
+  if (drawTimeout) clearTimeout(drawTimeout);
+  drawTimeout = setTimeout(draw, 1000);
+
+  const now = Date.now();
+
+  const elapsedTime = formatTime(totalTime + (state === RunState.RUNNING ? now - lastResumeTime : 0));
+
+  let pace: string;
+  if (now - lastGPS <= GPS_TIMEOUT_MS) {
+    pace = calculatePace(thisSplitTime(), splitDist).toFixed(2);
+  }else{
+    pace = "No GPS";
+  }
+
+  layout["time"]!.label = elapsedTime;
+  layout["pace"]!.label = pace;
+  layout.render();
+
+  if (now - lastUnlazy > 30000)
+    layout.forgetLazyState();
```

Comment:
I was wondering if the layout's rects were accumulating, but this doesn't resolve the leak